gapic/src/main/com/google/gapid/perfetto/models/FrameEventsTrack.java

```diff
@@ -333,6 +388,62 @@ public void getRange(Consumer<TimeSpan> span) {
         slice.getRange(span);
       }
     }
+
+    public FrameSelection getSelection() {
+      FrameSelection selection = new FrameSelection();
+      slices.forEach(s -> selection.combine(s.getSelection()));
+      return selection;
+    }
+  }
+
+  public static class FrameSelection {
+    public final List<Long> frameNumbers;
+    public final List<String> layerNames;
+    public static FrameSelection EMPTY = new FrameSelection();
+
+    public FrameSelection() {
+      frameNumbers = Lists.newArrayList();
+      layerNames = Lists.newArrayList();
+    }
+
+    public FrameSelection(long[] frameNumbers, String[] layerNames) {
+      this.frameNumbers = Lists.newArrayList(Longs.asList(frameNumbers));
+      this.layerNames = Lists.newArrayList(layerNames);
+    }
+
+    public boolean contains(long[] f, String[] l) {
+      for (int i = 0; i < f.length; i++) {
+        int idx = frameNumbers.indexOf(f[i]);
```

Comment:
This doesn't work if FrameSelection has multiple layers with the same frame number.

Reply:
The lines below this does that check

Reply:
I am thinking of the scenario where the layer name comes at a later index.  indexOf only returns the first one, the function will return false instead of true.

Reply:
Fixed it. Done.